packages/@aws-cdk/pipelines/lib/pipeline.ts

```diff
@@ -21,19 +23,28 @@ export interface CdkPipelineProps {
    *
    * @default - Required unless `codePipeline` is given
    */
-  readonly sourceAction?: codepipeline.IAction;
```

Comment:
I was thinking of a backward-compatible way but I think this will make the code unreadable. The reason for dev preview is to experiment and break APIs when needed imho.